lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -464,9 +722,805 @@ impl EnsoLexer {
+            _                                  => nested_text.style
+        };
+        nested_text.style         = new_style;
+        let literal               = nested_text.into();
+        let interpolate_is_closed = true;
+        self.append_token(literal);
+        self.on_interpolate_end(interpolate_is_closed,reader);
+    }
+
+    /// Triggered when ending an interpolation in a format text literal.
+    fn on_interpolate_end<R:LazyReader>(&mut self, is_closed:bool, reader:&mut R) {
+        trace!(self.logger,"Text::on_interpolate_end");
+        let text_interpolate   = self.text_interpolate;
+        let nested_format_line = self.text_format_line_nested;
+        if self.is_inside_state(text_interpolate) {
+            if self.is_inside_state(nested_format_line) {
+                self.pop_states_until(nested_format_line);
+                let current   = self.text_state.unsafe_current_mut();
+                current.style = Some(token::TextStyle::UnclosedLine);
+                let text      = self.text_state.unsafe_end_literal();
+                let literal   = text.into();
+                self.append_token(literal);
+            }
+            self.pop_states_until(text_interpolate);
+            let expr_tokens  = self.consume_tokens();
+            let offset       = self.offset.consume();
+            let expr_segment = if is_closed {
+                Token::TextSegmentInterpolate(expr_tokens.into(),offset)
+            } else {
+                Token::TextSegmentUnclosedInterpolate(expr_tokens.into(),offset)
+            };
+            self.text_state.append_segment(expr_segment);
+            self.pop_tokens();
+            self.offset.pop();
+            self.pop_state();
+        } else {
+            self.on_unrecognized(reader);
+        }
+    }
+
+    /// Triggered when encountering an EOF inside an interpolation.
+    fn on_eof_in_interpolate<R:LazyReader>(&mut self, reader:&mut R) {
+        trace!(self.logger,"Text::on_eof_in_interpolate");
+        let interpolate_is_closed = false;
+        self.on_interpolate_end(interpolate_is_closed,reader);
+        let mut text = self.text_state.unsafe_end_literal();
+        text.style   = match text.style {
+            Some(token::TextStyle::FormatLine) => Some(token::TextStyle::UnclosedLine),
+            _                                  => text.style,
+        };
+        let token = text.into();
+        self.append_token(token);
+        self.on_eof(reader);
+    }
+
+    /// Triggered when encountering a newline inside an interpolation.
+    fn on_newline_in_interpolate<R:LazyReader>
+    ( &mut self
+      , line_ending : token::LineEnding
+      , reader      : &mut R
+    ) {
+        trace!(self.logger,"Text::on_newline_in_interpolate");
+        let interpolate_is_closed = false;
+        self.on_interpolate_end(interpolate_is_closed,reader);
+        let current_style = self.text_state.unsafe_current().style.expect("Must be set.");
```

Comment:
The string `"Must be set."` repeats several times in this file. If we hae some repeating constants, maybe its worth refactoring them out?